x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -584,6 +584,10 @@ static Builder builder() {
         return new Builder();
     }
 
+    public DataType noText() {
```

Comment:
`textAsKeyword()`? `noText` is compact, but I find it a tad misleading. Minor preference, can stay as is.

Reply:
The idea is sound, but I have a minor preference for `noText` as `textAsKeyword` also seems to say something about the current datatype being TEXT; which it likely is not.